src/containers/Login/Login.jsx

```diff
@@ -189,8 +272,44 @@ class Login extends Component {
         </div>
         {signup && <Signup onSignup={this.handleSignup} onClose={this.handleClose} />}
         {syncing && <SyncDevice onClose={this.handleClose} />}
-        {addProfile && <AddProfile onClose={this.handleCloseAddProfile} onSignup={this.handleOpenSignup} onSync={this.handleOpenSync} />}
-        {sync && <SyncUrl onClose={this.handleCloseSyncUrl} />}
+        {addProfile &&
+          <div>
+            <MediaQuery maxWidth={700}>
+              <AddProfile onOpenSyncMethod={this.handleOpenSyncMethod} onClose={this.handleCloseAddProfile} onSignup={this.handleOpenSignup} />}
+            </MediaQuery>
+            <MediaQuery minWidth={701}>
+              <AddProfile onOpenSyncMethod={this.handleOpenSync} onClose={this.handleCloseAddProfile} onSignup={this.handleOpenSignup} />}
+            </MediaQuery>
+          </div>}
+        {syncMethod && (
+          <SyncMethod
+            onOpenOnboardingCopyLink={this.handleOpenOnboardingCopyLink}
+            onOpenOnboardingQrCode={this.handleOpenOnboardingQrCode}
+            onClose={this.handleCloseSyncMethod}
+            onSync={this.handleOpenSync}
+          />)}
+        {showOnboardingCopyLink && (
+          <OnBoarding
+            width={337}
+            height={480}
+            onClose={this.handleCloseOnboardingCopyLink}
+            slidesContent={slidesContentCopyLink}
+          />)}
+        {showOnboardingCopyLinkFromSyncProfile && (
+          <OnBoarding
+            width={337}
+            height={480}
+            onClose={this.handleCloseOnboardingCopyLinkFromSyncProfile}
+            slidesContent={slidesContentCopyLink}
+          />)}
+        {showOnboardingQrCode && (
+          <OnBoarding
+            width={337}
+            height={480}
+            onClose={this.handleCloseOnboardingQrCode}
+            slidesContent={slidesContentQrCode}
```

Comment:
I think you can just choose which slidesContent you want to pass to Onboarding component ?
```js
const slides = showOnboardingQrCode ? slidesContentQrCode : slidesContentCopyLink
return (
    <OnBoarding
            width={337}
            height={480}
            onClose={this.handleCloseOnboarding}
            slidesContent={slidesContentQrCode}
    />)
)
```
and handleCloseOnboarding() can just set this.setState({ showOnboarding: true })